src/java/org/apache/cassandra/service/accord/EndpointMapping.java

```diff
@@ -18,119 +18,76 @@
 
 package org.apache.cassandra.service.accord;
 
-import java.net.Inet4Address;
-import java.net.InetAddress;
-import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
 
-import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableCollection;
 import com.google.common.collect.ImmutableMap;
-import com.google.common.primitives.Ints;
 
 import accord.local.Node;
+import accord.utils.Invariants;
 import org.apache.cassandra.locator.InetAddressAndPort;
-import org.apache.cassandra.locator.Replica;
 
-public class EndpointMapping
+class EndpointMapping implements AccordEndpointMapper
 {
-    static Node.Id endpointToId(InetAddressAndPort endpoint)
+    public static final EndpointMapping EMPTY = new EndpointMapping(0, ImmutableMap.of(), ImmutableMap.of());
+    private final long epoch;
+    private final ImmutableMap<Node.Id, InetAddressAndPort> idToEndpoint;
+    private final ImmutableMap<InetAddressAndPort, Node.Id> endpointToId;
```

Comment:
I think you want `com.google.common.collect.ImmutableBiMap`?  That makes sure the 2 can't drift